File: dashlane-project/Dashlane/src/main/java/com/dashlane/dagger/singleton/SharingModule.java
package com.dashlane.dagger.singleton;

import com.dashlane.ui.screens.fragments.userdata.sharing.center.SharingDataProvider;
import com.dashlane.ui.screens.fragments.userdata.sharing.center.SharingDataProviderImpl;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class SharingModule {

    @Provides
    @Singleton
    public SharingDataProvider provideSharingDataProvider(SharingDataProviderImpl provider) {
        return provider;
    }
}

File: dashlane-project/Dashlane/src/main/java/com/dashlane/dagger/singleton/SharingModule.kt
package com.dashlane.dagger.singleton

import com.dashlane.ui.screens.fragments.userdata.sharing.center.SharingDataProvider
import com.dashlane.ui.screens.fragments.userdata.sharing.center.SharingDataProviderImpl
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import javax.inject.Singleton

@Module
@InstallIn(SingletonComponent::class)
open class SharingModule {
    @Provides
    @Singleton
    open fun provideSharingDataProvider(provider: SharingDataProviderImpl): SharingDataProvider {
        return provider
    }
}